### PR TITLE
Use speed slider to control frameDelay in GameLab Animation Tab

### PR DESCRIPTION
**Pull request by caleybrock**
Updating to match the frameDelay and frameRate values in the spec:

"The values will vary from 60 frames (so animating once every 2 seconds at 30fps) to 1 frame (animating 30 times in one second at 30fps).  This means that the left side of the slider actually maps to a larger number (but a slower speed) than the right side."

Also, replace all animation.frameRates to use frameDelay, and add a migration to set the frameDelay for old animations.

### CODE REVIEW

**Comment by caleybrock** [comment]
It's very possible this is going to break the test you showed me for valid data, so I'm looking into that now!

File: apps/src/gamelab/AnimationPicker/AnimationPreview.jsx [hunk 1]
```diff
@@ -56,16 +56,17 @@ const AnimationPreview = React.createClass({
   },
 
   advanceFrame: function () {
-    // Never advance the frame if the frameRate is 0
-    if (this.props.animationProps.frameRate === 0) {
+    // Never advance the frame if the frameDelay is 0
+    if (this.props.animationProps.frameDelay === 0) {
```
Comment (islemaster) [inline apps/src/gamelab/AnimationPicker/AnimationPreview.jsx]: This seems wrong.  If `frameDelay` is zero, shouldn't we always advance the frame?  Maybe this should be checking against `Infinity`.

File: apps/src/gamelab/AnimationPicker/animationPickerModule.js [hunk 2]
```diff
@@ -140,7 +140,7 @@ function loadImageMetadata(sourceUrl, callback) {
       sourceSize: {x: image.width, y: image.height},
       frameSize: {x: image.width, y: image.height},
       frameCount: 1,
-      frameRate: 15
+      frameDelay: 4
```
Comment (islemaster) [inline apps/src/gamelab/AnimationPicker/animationPickerModule.js:143]: I don't think these are equivalent; at 30fps a frameDelay of 4 represents an animation frameRate of 7.5fps.  Intentional?

File: apps/src/gamelab/animationListModule.js [hunk 3]
```diff
@@ -172,6 +172,16 @@ export function setInitialAnimationList(serializedAnimationList) {
     };
   }
 
+  // Convert frameRates to frameDelays.
+  for (let key in serializedAnimationList.propsByKey) {
+    let animation = serializedAnimationList.propsByKey[key];
+    if (animation.frameDelay === undefined && animation.frameRate !== undefined) {
+      animation.frameDelay = Math.round(30 / animation.frameRate);
+    } if (animation.frameDelay === null) {
```
Comment (caleybrock) [inline apps/src/gamelab/animationListModule.js]: At first, I didn't think this was necessary, but I somehow ended up in a state with null so I wanted to have a default state.

Comment (islemaster) [reply apps/src/gamelab/animationListModule.js]: Did you mean for this to be an `else if` case?

Comment (islemaster) [reply apps/src/gamelab/animationListModule.js]: More generally, should we try to fill `frameDelay` if it is null _or_ undefined?  It seems like we want to make a strong assertion that every animation should have a nonzero frameDelay.  Also for `frameRate` it's good practice to check for what it is (number), not what it isn't (undefined). What if our check went more like this?

``` javascript
if (!animation.frameDelay) {
  if (typeof animation.frameRate === 'number' && !isNaN(animation.frameRate)) {
    animation.frameDelay = Math.round(30 / animation.frameRate);
  } else {
    animation.frameDelay = 2;
  }
}
```

File: apps/src/gamelab/AnimationPicker/animationPickerModule.js (hunk 2 above)
Comment (caleybrock) [reply apps/src/gamelab/AnimationPicker/animationPickerModule.js:143]: Updated to 2

**Comment by islemaster** [comment]
LGTM!

**Pull request merged**